### PR TITLE
Fix patient encounter query

### LINKED ISSUE #1101: Encounters for the wrong patient appearing
This could be a just due to dodgy demo data, but worth looking into:

To reproduce (on feature branch, using demo data (the one with patient "Tina Ling" already created):

- Create and save new patient
- Go to the "Encounters" tab
- Notice the list of existing encounters for Tina Ling, not the newly created patient.

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #1101 

I broke it when doing the hooks refactoring.

### CODE REVIEW

File: client/packages/system/src/Patient/Encounter/ListView.tsx [hunk 1]
```diff
@@ -28,6 +29,12 @@ const EncounterListComponent: FC = () => {
   const { queryParams } = useUrlQueryParams({
     initialSort: { key: 'startDatetime', dir: 'desc' },
   });
+  const patientId = usePatient.utils.id();
+  const filterBy = queryParams.filterBy ?? {};
+  filterBy['patientId'] = {
+    equalTo: patientId,
+  };
+  queryParams.filterBy = filterBy;
```
Comment (CarlosNZ) [inline client/packages/system/src/Patient/Encounter/ListView.tsx:37]: Can you not follow the same pattern as the other ListViews and just include the filter in the main query?:
```ts
const { data, isError, isLoading } = useEncounter.document.list({
    ...queryParams,
    sortBy: {
      key: sortBy.key as EncounterSortFieldInput,
      isDesc: sortBy.isDesc,
      direction: sortBy.isDesc ? 'desc' : 'asc',
    },
    filterBy: { patientId: {equalTo: patientId } }
  });
```

Comment (CarlosNZ) [reply client/packages/system/src/Patient/Encounter/ListView.tsx:37]: Also, probably not a good idea to mutate the `queryParams` object directly, that should only ever be set by the URL (I think)

**Review by CarlosNZ — APPROVED** [review]
Looks good, thanks. Just one suggestion/query for consistency sake.

File: client/packages/system/src/Patient/Encounter/ListView.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/system/src/Patient/Encounter/ListView.tsx]: I didn't want to overwrite a potential filter from the query params but just add the patient filter on top...

But yes `queryParams` shouldn't be mutated I, I make a shallow copy now, thanks!

Comment (CarlosNZ) [reply client/packages/system/src/Patient/Encounter/ListView.tsx]: > I didn't want to overwrite a potential filter from the query params but just add the patient filter on top...

You could still do
```ts
   filterBy: { ...queryParams.filterBy, patientId: {equalTo: patientId } }
```
within the query?

But what you've got works too.

Comment (clemens-msupply) [reply client/packages/system/src/Patient/Encounter/ListView.tsx]: Agree this is simpler will go with your version, thanks!

**Pull request merged**